### DISCUSSION TITLE
no .xcodeproj file generated

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1935] [Q&A]
How do I open the project in xcode for iOS? there's no .xcodeproj file generated.

**Comment by rh101** [comment]
> How do I open the project in xcode for iOS? there's no .xcodeproj file generated.

If you follow this guide, then it should generate the project for you:
https://github.com/axmolengine/axmol/blob/dev/docs/DevSetup.md#ios-tvos-and-macos

If you are getting errors, then include the errors in your post.

**Comment by paulocoutinhox** [comment]
> How do I open the project in xcode for iOS? there's no .xcodeproj file generated.

I have some shortcuts in Makefile that can help you (ios, macos, tvos):
https://github.com/paulocoutinhox/axmol-native-helper/blob/main/Makefile

**Comment by meehjeevan** [comment]
While trying to run on iOS simulator I got this error:
![Screen Shot 2024-05-29 at 3 25 52 PM](https://github.com/axmolengine/axmol/assets/6400139/caa0c911-33b6-4137-9a37-d0cc15fbdab9)

**Reply by meehjeevan** [reply]
And sometime these errors too. 
![Screen Shot 2024-05-29 at 5 17 37 PM](https://github.com/axmolengine/axmol/assets/6400139/4c5e32f4-e37a-4073-8c3d-ab857efe45a4)
![Screen Shot 2024-05-29 at 5 17 00 PM](https://github.com/axmolengine/axmol/assets/6400139/88521ae5-253b-4153-bf1e-4f21e18ae3db)

**Reply by rh101** [reply]
Show the step by step process you used to create project, and also list the environment, which OS version, Xcode version, CMake version, etc..

**Reply by meehjeevan** [reply]
for creating project I've used the same command from the guide. 
Android version is working.. 
for ios version, I ran this command
axmol build -p ios -a x64 -c

build_ios_x64 folder was generated, from there I opened the .xcodeproj file on xcode and tried running on simulator

OS version: 10.15.7
Xcode: 12.4
cmake: 3.29.3

**Reply by rh101** [reply]
I think the docs are out of date. Try this instead: 
(arm64 Sim) `axmol build -p ios -a arm64 -sdk simulator -c` 
or 
(x64 Sim) `axmol build -p ios -a x64 -sdk simulator -c`

**Comment by meehjeevan** [comment]
same issue with this too... 
is it because I'm using an old mac and max version of xcode I can use for now is 12.4.

**Reply by meehjeevan** [reply]
any solutions yet ?

**Reply by rh101** [reply]
> is it because I'm using an old mac and max version of xcode I can use for now is 12.4.

It could be, but, have you tested this with a new axmol project? If not, please create a new project, then attempt to build that.

**Reply by meehjeevan** [reply]
Yes, new blank project runs on simulator  but the whole project which is working well on android is not working for ios. 
What do you suggest to make it work?
And should I run this command axmol build -p ios -a arm64 -sdk simulator -c each time I make changes in the project? on addition or deletion of cpp files or any other resources?

**Reply by rh101** [reply]
> Yes, new blank project runs on simulator but the whole project which is working well on android is not working for ios.
> What do you suggest to make it work?

Sorry, I have absolutely no idea what you are asking about; you have provided no information about what exactly is not working.

> And should I run this command axmol build -p ios -a arm64 -sdk simulator -c each time I make changes in the project? on addition or deletion of cpp files or any other resources?

Yes, any changes you make to the `CMakeLists.txt`, any source changes (if using `GLOB` in `CMakeLists.txt`), or any resources changed in the `Content` folder etc., would require you to regenerate the Xcode project.

**Reply by meehjeevan** [reply]
ok, let me explain. I have a project in cocos2dx 3.17.2 which I migrated to Axmol. And it's working fine on Android. 
Now, I have to build for ios but when I open the project in xcode and hit run then a lot of errors appears. Please check the screenshots.  
<img width="1055" alt="Screenshot 2024-08-13 at 8 28 43 PM" src="https://github.com/user-attachments/assets/885ae538-7b7f-461d-b6d4-8d439f94ae6e">
<img width="1053" alt="Screenshot 2024-08-13 at 8 27 41 PM" src="https://github.com/user-attachments/assets/24df6d98-8ad0-43b5-ad64-ed896a8e54ba">
<img width="1043" alt="Screenshot 2024-08-13 at 8 27 17 PM" src="https://github.com/user-attachments/assets/b7875516-bd8e-4940-a04a-d0e46004dc8c">

**Reply by rh101** [reply]
Errors like that are usually a result of a mistake in the project source code.

Have you tested a newly generated project (with no modifications) using Xcode to confirm if it works or not?  From your post above, it's not clear if you have attempted that. If a new project builds with no issues on Xcode, then you can be sure it's a problem in the project ported from Cocos2d-x, in which case you would have to carefully go over the entire project to find the error.

How did you migrate the project across to Axmol? Did you generate a new project using Axmol, then port the source code over, or did you modify the original cocos2d-x project to get it building with Axmol? If it's the latter, then I suggest you generate a new project in Axmol, and then carefully move all the source code across to it.  If there are modifications you have made to the `CMakeLists.txt` in the original project, then review them carefully before porting them over to the new project.

Also show the each and every step of how you generate the project, such as the `axmol new xyz` and `axmol build xyz` commands.

**Reply by meehjeevan** [reply]
Yes, I have tried running a newly generated project on xcode and it works. 
While migrating to Axmol, yes, I generated new project in axmol with the command
`axmol new -p packageId -l cpp projectName `

And I placed classes and resources files into the newly generated project, I ran & fixed the class file one by one according to the errors shown while running on Android Studio. It took a lot of time to get it work for android. But now same migrated project is showing such errors while trying to build for ios.

**Reply by rh101** [reply]
> It took a lot of time to get it work for android. But now same migrated project is showing such errors while trying to build for ios.

Then the issue is likely to be in your source files.  You stated that your project builds for Android, so just going off that, it may mean that there are iOS-specific sections of code or source files in your project that may be causing the problem for you, so that's where you may want to start looking.

**Reply by smilediver** [reply]
Are you still using Xcode: 12.4? Maybe that is the problem. Also, are you by any chance including Axmol headers into `.c` or `.m` files? Those errors suspiciously look like being compiled by C compiler instead of C++.

**Reply by meehjeevan** [reply]
No, I'm using xcode 15.4. I got it worked in a quite strange way. First I ran the newly generated project on xcode and then I placed class files (.cpp and .h) one by one in the source folder and in cmakelists and hit run. Whichever file it threw error as not found I placed them one by one and kept hitting run then it finally worked. There wasn't really errors in codes. It just didn't work at once with dozens of class files. I don't know how migrated project on ios got it worked for you guys but I got it worked this way.
Funny thing, till now, I was running on simulator. And when I now tried to run on connected device then again same error occurred. Again have to do repeat the same cycle. 
BTW, for real device is this command correct? 
`axmol build -p ios -a arm64 -c`
If yes, Should I use same command for release version too?

**Reply by rh101** [reply]
> BTW, for real device is this command correct?
> `axmol build -p ios -a arm64 -c`
> If yes, Should I use same command for release version too?

Yes, that's the correct command to generate the project for real devices.  The same project is used for any type of build, debug, release, release with debug info etc..  Once you open the project in Xcode, set the type of build in the scheme settings.

**Reply by meehjeevan** [reply]
Hello everyone,

I’m able to run the app on a connected device, but I encountered the same errors shown in the screenshots when trying to archive it in Xcode.

I tried archiving on an empty, blank project that is generated initially when creating a project using the command, and archiving works there. However, when I replace the source, contents, and CMakeLists.txt with my project’s files, I can run the app on a connected device, but archiving still doesn’t work.
Same errors are shown when I run Analyze on xcode too. 
Now, I'm unable to create .ipa file and can't distribute or publish update on app store. What would I do now?

**Reply by smilediver** [reply]
Do you have `.c` or `.m` files in your project? Do they include any Axmol headers? Do you include Axmol headers into prefix/precompiled header? I think you have C or Objective-C files that include Axmol's C++ headers, and hence the compilation errors.

You can use the following method to export and post your build log, or at least find and post the logs for failing commands, that would make it easier to determine the build issues: https://stackoverflow.com/a/30084344/254101.

**Reply by meehjeevan** [reply]
No, I don't have any .c or .m files created by myself. I figured it out, I have a file which has same name and it conflicted with a file in ios sdk's coreFoudation. Thanks.
